pleskdistup/actions/common_checks.py

```diff
@@ -142,3 +144,109 @@ def __init__(self):
 
     def _do_check(self) -> bool:
         return subprocess.run(["/bin/fuser", "/var/lib/apt/lists/lock"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL).returncode != 0
+
+
+class MinFreeDiskSpaceViolation(typing.NamedTuple):
+    """Information about a filesystem with insufficient free disk space."""
+    dev: str
+    """Device name."""
+    req_bytes: int
+    """Required space on the device (in bytes)."""
+    avail_bytes: int
+    """Available space on the device (in bytes)."""
+    paths: typing.Set[str]
+    """Paths belonging to this device."""
+
+
+class AssertMinFreeDiskSpace(action.CheckAction):
+    """Check if there's enough free disk space.
+
+    Args:
+        requirements: A dictionary mapping paths to minimum free disk
+            space (in bytes) on the devices containing them.
+        name: Name of the check.
+        description: Description to show if the assertion is violated. If ends
+            with a colon, list of violations will be added automatically.
+    """
+    violations: typing.List[MinFreeDiskSpaceViolation]
+    """List of filesystems with insiffucient free disk space."""
+    _description: str
+
+    def __init__(
+        self,
+        requirements: typing.Dict[str, int],
+        name: str = "check if there's enough free disk space",
+        description: str = "There's not enough free disk space:",
+    ):
+        self.requirements = requirements
+        self.name = name
+        self._description = description
+        self.violations = []
+
+    @property
+    def description(self) -> str:
+        """Description of violations or empty string.
+
+        Can be set to change the message template.
+        """
+        if not self.violations:
+            return ""
+        res = self._description
```

Comment:
Suggestion: Do we really need to be able to change `_description` at all? Perhaps we could simply hard-code the initial part here. This way, we can remove the setter and no longer need to check `if res.endswith(":"):`.

Reply:
Unfortunately, we can't have just a getter without a setter, because `description` is a base class attribute and you can't override it with non-data descriptor. You will get "Cannot override writeable attribute with read-only property" from mypy. I will just drop the property and assign the attribute directly.

Reply:
Yes, I think it would be better to simply return `_description` in the getter. and set the description from `_do_check`. Additionally, it's not necessary to store valuations in the checker class in this case, which means we store less data and have fewer problems.

Reply:
I'm for results accessibility in machine-readable form, not only human-readable strings.

Reply:
> I'm for results accessibility in machine-readable form, not only human-readable strings.

Hm, it might be better to make it possible to return pre-checker results in json (or any similar format)

Reply:
At least it would be nice if checks would provide access to unserialized results for potential use in other parts of the program, e.g. for testing.

Reply:
> e.g. for testing.

I believe in this case you should split function that takes results and test this new function.